src/mlpack/core/data/split_data.hpp

```diff
@@ -179,13 +182,15 @@ void StratifiedSplit(const arma::Mat<T>& input,
  * @param shuffleData If true, the sample order is shuffled; otherwise, each
  *       sample is visited in linear order. (Default true.)
  */
-template<typename T, typename U>
+template<typename T, typename LabelsType,
+         typename = std::enable_if_t<arma::is_arma_type<LabelsType>::value ||
+                          arma::is_Mat_only<LabelsType>::value> >
```

Comment:
Isn't the `is_Mat_only` redundant now?

Reply:
Yes. You are right. I simply did a find and replace for this. I should have paid a closer look :-)